compat/ioapi.h

```diff
@@ -5,7 +5,14 @@
 #include <stdint.h>
 
 #ifndef _ZLIB_H
-//#include "zlib.h"
+#include "zlib.h"
+#endif
+
+#ifndef Z_ERRNO
+#define Z_ERRNO                         (-1)
+#endif
+#ifndef Z_DEFLATED
+#define Z_DEFLATED                      (8)
 #endif
```

Comment:
I can also remove this entirely.
After all, we don't attempt to redefine `Z_DEFAULT_STRATEGY` which we use and which is also coming from zlib.h